### PR TITLE
Skid landing take #2

### PR DESCRIPTION
**Pull request by podhrmic**
Fixes from  #1656
I renumbered (sequentially) the conf examples so it is easier to add more examples in the future (I had a hard time finding an unused ID).

### CODE REVIEW

File: sw/airborne/modules/nav/nav_skid_landing.c [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+#ifndef SKID_LANDING_FINAL_HEIGHT //> final height [m]
+#define SKID_LANDING_FINAL_HEIGHT 5
+#endif
+#ifndef SKID_LANDING_FINAL_STAGE_TIME //> final stage time [s]
+#define SKID_LANDING_FINAL_STAGE_TIME 5
+#endif
+
+static inline float distance_equation(struct FloatVect2 p1,struct FloatVect2 p2)
+{
+  return sqrt((p1.x-p2.x)*(p1.x-p2.x)+(p1.y-p2.y)*(p1.y-p2.y));
```
Comment (flixr) [inline sw/airborne/modules/nav/nav_skid_landing.c]: for float values use `sqrtf` or even directly something like
```
struct FloatVect2 diff;
VECT2_DIFF(diff, p1, p2);
return float_vect2_norm(diff);
```
or even add a distance function in math?

**Comment by flixr** [comment]
Please add the new defines in the module xml documentation section

**Comment by flixr** [comment]
Like @gautierhattenberger I don't really like adding new modes like this, but currently don't see any nice alternative...

**Comment by flixr** [comment]
regarding renumbering the AC_IDs in conf_tests: I think it is good to keep the same AC_IDs for the aircrafts as in conf_example

**Comment by flixr** [comment]
Also could you please describe what the benefits of this skid landing module are compared to the usual flight plan flare landing (easier usage is great, but what are the behavior/performance differences)?

File: sw/airborne/modules/nav/nav_skid_landing.c (hunk 1 above)
Comment (podhrmic) [inline sw/airborne/modules/nav/nav_skid_landing.c]: Changed to sqrtf

**Comment by podhrmic** [comment]
@flixr to your comments:

> Please add the new defines in the module xml documentation section

it is already there? https://github.com/paparazzi/paparazzi/pull/1669/files#diff-8fa01c82a9a3fc563b913bc4d4ebb5b2
I am trying to group the defines in a section (i.e. ` <section name="LANDING" prefix="SKID_LANDING_">` so it is more organized. How else do you want it?

 

> regarding renumbering the AC_IDs in conf_tests: I think it is good to keep the same AC_IDs for the aircrafts as in conf_example

Should I renumber the ACs in conf_example too? There is no reason for having them randomly numbered - plus now you at least know what IDs are free so you can add more examples/tests (like I did).

> Also could you please describe what the benefits of this skid landing module are compared to the usual flight plan flare landing (easier usage is great, but what are the behavior/performance differences)?

Let me try:

1. It is an alternative to the existing landing procedure that is defined in the flight plan (for example [here](https://github.com/paparazzi/paparazzi/blob/master/conf/flight_plans/versatile.xml#L126)
2. it is easier to use and gives you consistent behavior - because you only call setup/run fucntions instead of defining the landing procedure in the flight plan (so you can't make a mistake there)
3. It has been used for many landings during various missions so it is field proven.
4. I think it behaves similarly (i.e. circle down, follow a line, flare), it is just nicely wrapped in a module.

**Comment by flixr** [comment]
Regarding the docs for defines: since you commented that part it will not be part of the generated docs.
If you add the `<section>` to the doc node (and add a description for each define) it will also be part of the generated docs, like e.g. in http://docs.paparazziuav.org/latest/module__optical_flow_landing.html

> Should I renumber the ACs in conf_example too?

IMHO that would be preferable to only renumbering them in one...

Thanks for the explanation, it's just that from your previous comment "improved flare" sounded like it does it a bit better/differently...

**Comment by flixr** [comment]
Regarding the V_CTL_LANDING gains: they should probably default to the corresponding V_CTL_AUTO gains?

**Comment by podhrmic** [comment]
> Regarding the V_CTL_LANDING gains: they should probably default to the corresponding V_CTL_AUTO gains?

Some gains can be used from V_CTL_AUTO gains, but what about for example V_CTL_LANDING_DESIRED_SPEED or V_CTL_LANDING_ALT_FLARE? That has no counterpart in the V_CTL_AUTO world - and since it heavily depends on the airframe used there seem to be no good default values.

Should I 

1. define them as zero which will make the V_CTL_LANDING not really useful unless you define your own gains (and have a warning or so printed in such case)>
2. keep them on some default value that is probably not great, but at least it will somewhat work with default gains (and again print a warning)
3. dont define them, so if the user doesn't define the proper gains the code wont compile?

**Comment by flixr** [comment]
I didn't look at the landing control loops in detail, but if they are basically the same as the "normal" loops, it would probably make sense to default to those gains as they work (whereas the fixed default might not).
For the DESIRED_SPEED and ALT_FLARE, either way is fine with me... provide a default or require it to be defined.

**Comment by flixr** [comment]
Also what happens if someone tries to use this skid landing module with the new/adaptive control loops?

**Comment by podhrmic** [comment]
Well the skid landing is using the new mode V_CTL_LANDING - and at least for our aircrafts the gains are different. V_CTL_LANDING controls the pitch during flare, and I guess that because that is outside of normal flight envelope (close to the stall speed, high angle of attack), the normal flight gains won't work very well. 

For adaptive control - it wont compile, because the adaptive controls don't have V_CTL_LANDING

**Comment by flixr** [comment]
Well, if you think that users should provide separate gains for V_CTL_LANDING, then maybe it doesn't make much sense to have defaluts (for all those defines)?

**Comment by flixr** [comment]
But if the flight plan flare landing works well with the normal gains and someone just wants to use this module to make things easier, it would make sense to default...

**Comment by podhrmic** [comment]
I ll look into it.

File: conf/modules/nav_skid_landing.xml [hunk 2]
```diff
@@ -19,16 +19,54 @@ Block example:
        <define name="FINAL_HEIGHT" value="50" unit="m"/>
        <define name="FINAL_STAGE_TIME" value="10" unit="s"/>
      </section>
-</block>
 @endverbatim
 -->
+
+Also define 
+<!--
```
Comment (flixr) [inline conf/modules/nav_skid_landing.xml]: if you comment these things out, they won't appear in the generated docs: http://docs.paparazziuav.org/latest/module__nav_skid_landing.html

You can instead make the complete `<section>` part of the `<doc>` node...

Comment (podhrmic) [inline conf/modules/nav_skid_landing.xml]: fixed with 6c19fe313d59775528ffc6fd5b3083022cc35cbf

**Pull request merged**